src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/OffenderService.kt

```diff
@@ -75,7 +77,13 @@ class OffenderService(
   }
 
   fun getOffenderByCrn(crn: String, userDistinguishedName: String): AuthorisableActionResult<OffenderDetailSummary> {
-    val offender = when (val offenderResponse = communityApiClient.getOffenderDetailSummary(crn)) {
+    val offenderResponse = if (offenderRepository.offenderHasApplicationOrBooking(crn)) {
+      communityApiClient.getOffenderDetailSummaryWithWait(crn)
+    } else {
+      communityApiClient.getOffenderDetailSummaryWithCall(crn)
+    }
```

Comment:
Good point, for this use case (offener/inmate details) a CRN search should always come before a Booking or Application is created so it makes sense to limit the number of polls to 1 (by just setting the timeout to 0s) because the cache should never be cold except at first deploy or after a deliberate cache clear.  Have made the change.